src/crystal/system/win32/time.cr

```diff
@@ -1,9 +1,47 @@
+require "c/winbase"
+
 module Crystal::System::Time
+  # Win32 epoch is 1601-01-01 00:00:00 UTC
+  WINDOWS_EPOCH_IN_SECONDS = 50_491_123_200_i64
+
+  # Resolution of FILETIME is 100 nanoseconds
+  NANOSECONDS_PER_FILETIME_TICK = 100
+
+  NANOSECONDS_PER_SECOND    = 1_000_000_000
+  FILETIME_TICKS_PER_SECOND = NANOSECONDS_PER_SECOND / NANOSECONDS_PER_FILETIME_TICK
+
+  # TODO: For now, this method returns the UTC offset currently in place, ignoring *seconds*.
   def self.compute_utc_offset(seconds : Int64) : Int32
-    raise NotImplementedError.new("Crystal::System::Time.compute_utc_offset")
+    ret = LibC.GetTimeZoneInformation(out zone_information)
+    raise Errno.new("GetTimeZoneInformation") if ret == -1
+
+    zone_information.bias.to_i32 * -60
   end
 
   def self.compute_utc_seconds_and_nanoseconds : {Int64, Int32}
-    raise NotImplementedError.new("Crystal::System::Time.compute_utc_seconds_and_nanoseconds")
+    LibC.GetSystemTimePreciseAsFileTime(out filetime)
```

Comment:
In the future we should use windows's `dlopen` equivalent to fallback to `GetSystemTimeAsFileTime`. Can we either do that in this PR (see [this](https://github.com/dotnet/coreclr/pull/9736/files#diff-0d2e4fca23d725bbd5e637d1fe57ed0fR52)) or leave a TODO?

Reply:
@RX14 that approach will be required if we want to support anything older than Windows 8 (function was introduced then).

Considering a few other places might require such approach, perhaps adding a TODO will be recommended.

Cheers.

Reply:
Should this check be at the callsite or in the bindings definition?
But I think I'd rather leave the implementation out of this PR.

Reply:
The dlopen and method exists check would want to be cached, yes.